src/typecheck/mod.rs

```diff
@@ -1314,6 +1314,52 @@ fn row_add(
     }
 }
 
+/// Get the set of all unification variables in a type.
+fn get_free_variables(wildcards: &Vec<TypeWrapper>, t: TypeWrapper) -> HashSet<usize> {
```

Comment:
Yes, I initially wanted to do that. But I thought I'd just write this once, because there might be other places which might need to get all the free unification variables.

Reply:
I think I'd rather create an inner function that threads a mutable argument

Reply:
Do you have anything in mind in particular that would need the free unification variables? If yes, then go. Otherwise, I think you probably aren't gonna need it :slightly_smiling_face: and I would implement the hashset-free version. If really we were wrong and need this later, this isn't very hard to move to the hashset version.

Reply:
I don't think I have, I'll go with the check now